frontend/src/stimulus/controllers/dynamic/work-packages/activities-tab/index.controller.ts

```diff
@@ -0,0 +1,422 @@
+
+    this.openEditorWithInitialData(this.quotedText(content, userName));
+  }
+
+  private quotedText(rawComment:string, userName:string) {
+    const quoted = rawComment.split('\n')
+      .map((line:string) => `\n> ${line}`)
+      .join('');
+
+    return `${userName}\n${quoted}`;
+  }
+
+  openEditorWithInitialData(quotedText:string) {
+    this.showForm();
+    const ckEditorInstance = this.getCkEditorInstance();
+    if (ckEditorInstance && ckEditorInstance.getData({ trim: false }).length === 0) {
+      ckEditorInstance.setData(quotedText);
+    }
+  }
+
+  clearEditor() {
+    this.getCkEditorInstance()?.setData('');
+  }
+
+  hideEditorIfEmpty() {
+    const ckEditorInstance = this.getCkEditorInstance();
+    if (ckEditorInstance && ckEditorInstance.getData({ trim: false }).length === 0) {
+      this.buttonRowTarget.classList.remove('d-none');
+      this.formRowTarget.classList.add('d-none');
+
+      if (this.journalsContainerTarget) {
+        this.journalsContainerTarget.style.marginBottom = '';
+        this.journalsContainerTarget.classList.add('with-initial-input-compensation');
+        this.journalsContainerTarget.classList.remove('with-input-compensation');
+      }
+    }
+  }
+
+  async onSubmit(event:Event | null = null) {
+    event?.preventDefault();
+    const ckEditorInstance = this.getCkEditorInstance();
+    const data = ckEditorInstance ? ckEditorInstance.getData({ trim: false }) : '';
+
+    const formData = new FormData(this.formTarget);
+    formData.append('last_update_timestamp', this.lastUpdateTimestamp);
+    formData.append('filter', this.filterValue);
+    formData.append('journal[notes]', data);
+
+    const response = await this.fetchWithCSRF(this.formTarget.action, 'POST', formData);
+
+    if (response.ok) {
+      this.setLastUpdateTimestamp();
+      const text = await response.text();
+      Turbo.renderStreamMessage(text);
+
+      if (this.journalsContainerTarget) {
+        this.clearEditor();
+        if (this.isMobile()) {
+          this.hideEditorIfEmpty();
+        } else {
+          this.focusEditor();
+        }
+        if (this.journalsContainerTarget) {
+          this.journalsContainerTarget.style.marginBottom = '';
+          this.journalsContainerTarget.classList.add('with-input-compensation');
+        }
+        setTimeout(() => {
+          this.scrollJournalContainer(
+            this.journalsContainerTarget,
+            this.sortingValue === 'asc',
+          );
+          if (this.isMobile()) {
+            this.scrollInputContainerIntoView(300);
+          }
+        }, 10);
+      }
+    }
+  }
+
+  private async fetchWithCSRF(url:string | URL, method:string, body?:FormData) {
+    return fetch(url, {
+      method,
+      body,
+      headers: {
+        'X-CSRF-Token': (document.querySelector('meta[name="csrf-token"]') as HTMLMetaElement).content,
+        Accept: 'text/vnd.turbo-stream.html',
+      },
+      credentials: 'same-origin',
+    });
+  }
```

Comment:
We should extract this into the  TurboRequests service (src/app/core/turbo/turbo-requests.service.ts) , except for the token, it's already there. It's still valid to use Angular services for this use case.